packages/ui/src/validators/components/ValidatorsList.tsx

```diff
@@ -1,85 +1,90 @@
-import React, { useEffect, useState } from 'react'
+import React, { useMemo, useState } from 'react'
+import { generatePath } from 'react-router-dom'
 import styled from 'styled-components'
 
 import { List, ListItem } from '@/common/components/List'
 import { ListHeader } from '@/common/components/List/ListHeader'
 import { SortHeader } from '@/common/components/List/SortHeader'
 import { Tooltip, TooltipDefault } from '@/common/components/Tooltip'
 import { Colors } from '@/common/constants'
+import { Comparator } from '@/common/model/Comparator'
+import { WorkingGroupsRoutes } from '@/working-groups/constants'
 
-import { Validator } from '../types'
+import { ValidatorWithDetails } from '../types'
 
 import { ValidatorItem } from './ValidatorItem'
 
 interface ValidatorsListProps {
-  validators: Validator[]
+  validators: ValidatorWithDetails[]
 }
 
 export const ValidatorsList = ({ validators }: ValidatorsListProps) => {
-  const [sortByApr, setSortByApr] = useState<'asc' | 'desc' | undefined>(undefined)
-  const [sortByCommission, setSortByCommission] = useState<'asc' | 'desc' | undefined>(undefined)
-  const [sortedValidators, setSortedValidators] = useState<Validator[]>([])
+  type SortKey = 'stashAccount' | 'APR' | 'commission'
+  const [sortBy, setSortBy] = useState<SortKey>('stashAccount')
+  const [isDescending, setDescending] = useState(false)
 
-  useEffect(() => {
-    setSortedValidators(validators)
-  }, [validators])
-
-  useEffect(() => {
-    if (!sortByApr) return
-    setSortedValidators(sortedValidators.sort((a, b) => (sortByApr === 'desc' ? a.APR - b.APR : b.APR - a.APR)))
-  }, [sortByApr])
+  const sortedValidators = useMemo(
+    () =>
+      [...validators].sort(
+        Comparator<ValidatorWithDetails>(isDescending, sortBy)[sortBy === 'stashAccount' ? 'string' : 'number']
+      ),
+    [sortBy, isDescending, validators]
+  )
 
-  useEffect(() => {
-    if (!sortByCommission) return
-    setSortedValidators(
-      sortedValidators.sort((a, b) =>
-        sortByCommission === 'desc' ? a.commission - b.commission : b.commission - a.commission
-      )
-    )
-  }, [sortByCommission])
+  const onSort = (key: SortKey) => {
+    if (key === sortBy) {
+      setDescending(!isDescending)
+    } else {
+      setDescending(false)
+      setSortBy(key)
```

Comment:
I realized that it makes more sense to see the APR and commission by descending order by default. But the validator names should ascending by default.

WDYT of:
```suggestion
  const onSort = (key: SortKey, descendingByDefault = false) => {
    if (key === sortBy) {
      setDescending(!isDescending)
    } else {
      setDescending(descendingByDefault)
      setSortBy(key)
```

Reply:
I agree.